koji_wrapper/tag.py

```diff
@@ -19,6 +19,20 @@ def __init__(self, tag=None,
         self.tagged_list = None
         super().__init__(**kwargs)
 
+    def __str__(self):
+        """return tag as description(str for human)"""
+        return self.tag
+
+    def __repr__(self):
+        """return detailed description string (str for machine)"""
+        params = []
+        for key in ['tag', 'blacklist', 'nvr_blacklist', 'profile',
+                    'user_config', 'url', 'topurl', 'session']:
+            value = self.__getattribute__(key)
```

Comment:
This seems to fail when attempting to get a key that doesn't exist. Are we absolutely guaranteed all these keys will always exist in a KojiTag object? If not, perhaps we should first check with hasattr()?

Reply:
all of these keys should be @property entries in KojiWrapperBase  KojiWrapper or KojiTag

Are you seeing failures in unittest or local testing or more the general that __getattribute__ will throw an error if key does not exists?

Reply:
More generally while I was poking around. I'm not super familiar with koji_wrapper yet but if all these properties are sure to be there then let's ignore that comment. Thanks for the explanation!